### PR TITLE
[FLINK-18320][notice][hive] Merge hive-exec dependencies for hive uber 2.3.6 and 3.1.2

### PR DESCRIPTION
**Pull request by JingsongLi**
## What is the purpose of the change

flink-sql-connector-hive depends on hive-exec which is a uber jar and doesn't come with a proper NOTICE file. So we need to find out the artifacts included in hive-execand list them in our NOTICE file.

## Brief change log

Update NOTICE file and add licenses for some dependencies.

### CODE REVIEW

**Comment by rmetzger** [comment]
I propose to review this PR only after https://github.com/apache/flink/pull/12682 has been resolved. It seems we are not entirely sure how to deal with the shaded hive-exec dependencies.

**Comment by JingsongLi** [comment]
> I propose to review this PR only after #12682 has been resolved. It seems we are not entirely sure how to deal with the shaded hive-exec dependencies.

Agree~

File: flink-connectors/flink-sql-connector-hive-3.1.2/src/main/resources/META-INF/NOTICE [hunk 1]
```diff
@@ -8,3 +8,51 @@ This project bundles the following dependencies under the Apache Software Licens
 
 - org.apache.hive:hive-exec:3.1.2
 - org.apache.thrift:libfb303:0.9.3
+
+The bundled Apache Hive org.apache.hive:hive-exec dependency bundles the following dependencies under
+the Apache Software License 2.0 (http://www.apache.org/licenses/LICENSE-2.0.txt)
+
+- com.esotericsoftware:kryo-shaded:3.0.3
+- com.esotericsoftware:minlog:1.3.0
+- com.esotericsoftware:reflectasm:1.10.1
```
Comment (rmetzger) [inline flink-connectors/flink-sql-connector-hive-3.1.2/src/main/resources/META-INF/NOTICE:17]: com.esotericsoftware:reflectasm is listed twice, and not ASL2.0 licensed.

File: flink-connectors/flink-sql-connector-hive-3.1.2/src/main/resources/META-INF/NOTICE [hunk 2]
```diff
@@ -8,3 +8,51 @@ This project bundles the following dependencies under the Apache Software Licens
...
+- org.codehaus.jackson:jackson-core-asl:2.9.5
+- org.codehaus.jackson:jackson-mapper-asl:2.9.5
+- org.objenesis:objenesis:2.1
+
+The bundled Apache Hive org.apache.hive:hive-exec dependency bundles the following dependencies under the BSD license.
+See bundled license files for details.
+
+- com.esotericsoftware:kryo-shaded:3.0.3
+- com.esotericsoftware:minlog:1.3.0
+- com.esotericsoftware:reflectasm:1.10.1
```
Comment (rmetzger) [inline flink-connectors/flink-sql-connector-hive-3.1.2/src/main/resources/META-INF/NOTICE:55]: where did you find this dependency listed? When I compiled hive 3.1.2, it wasn't listed by the shade plugin. However. there are indeed reflectasm classes in the final binary.

File: flink-connectors/flink-sql-connector-hive-3.1.2/src/main/resources/META-INF/NOTICE [hunk 3]
```diff
@@ -8,3 +8,51 @@ This project bundles the following dependencies under the Apache Software Licens
...
+- com.google.guava:guava:19.0
+- com.googlecode.javaewah:JavaEWAH:0.3.2
+- com.tdunning:json:1.8
+- commons-lang:commons-lang:2.6
+- io.airlift:aircompressor:0.10
+- javax.jdo:jdo-api:3.0.1
+- joda-time:joda-time:2.9.9
+- net.sf.opencsv:opencsv:2.3
+- org.apache.avro:avro-mapred:1.8.2
```
Comment (rmetzger) [inline flink-connectors/flink-sql-connector-hive-3.1.2/src/main/resources/META-INF/NOTICE:26]: [INFO] Including org.apache.avro:avro-mapred:jar:hadoop2:1.7.7 in the shaded jar.

File: flink-connectors/flink-sql-connector-hive-3.1.2/src/main/resources/META-INF/NOTICE [hunk 4]
```diff
@@ -8,3 +8,51 @@ This project bundles the following dependencies under the Apache Software Licens
...
+- org.apache.hive.shims:hive-shims-common:3.1.2
+- org.apache.hive:hive-common:3.1.2
+- org.apache.hive:hive-llap-client:3.1.2
+- org.apache.hive:hive-llap-common:3.1.2
+- org.apache.hive:hive-metastore:3.1.2
+- org.apache.hive:hive-serde:3.1.2
+- org.apache.hive:hive-service-rpc:3.1.2
+- org.apache.hive:hive-spark-client:3.1.2
+- org.apache.hive:hive-standalone-metastore:3.1.2
+- org.apache.hive:hive-storage-api:3.1.2
```
Comment (rmetzger) [inline flink-connectors/flink-sql-connector-hive-3.1.2/src/main/resources/META-INF/NOTICE:39]: I don't know what's going on with my maven

```
[INFO] Including org.apache.hive:hive-storage-api:jar:2.7.0 in the shaded jar.
```

File: flink-connectors/flink-sql-connector-hive-3.1.2/src/main/resources/META-INF/NOTICE [hunk 5]
```diff
@@ -8,3 +8,51 @@ This project bundles the following dependencies under the Apache Software Licens
...
+- org.apache.orc:orc-core:1.5.6
+- org.apache.orc:orc-shims:1.5.6
+- org.apache.orc:orc-tools:1.5.6
+- org.apache.parquet:parquet-hadoop-bundle:1.10.0
+- org.apache.parquet:parquet-jackson:1.10.0
```
Comment (rmetzger) [inline flink-connectors/flink-sql-connector-hive-3.1.2/src/main/resources/META-INF/NOTICE:44]: `parquet-jackson` was not listed in my hive build

File: flink-connectors/flink-sql-connector-hive-3.1.2/src/main/resources/META-INF/NOTICE [hunk 6]
```diff
@@ -8,3 +8,51 @@ This project bundles the following dependencies under the Apache Software Licens
...
+- org.apache.thrift:libthrift:0.9.3
+- org.codehaus.jackson:jackson-core-asl:2.9.5
+- org.codehaus.jackson:jackson-mapper-asl:2.9.5
```
Comment (rmetzger) [inline flink-connectors/flink-sql-connector-hive-3.1.2/src/main/resources/META-INF/NOTICE:47]:
```
[INFO] Including org.codehaus.jackson:jackson-core-asl:jar:1.9.13 in the shaded jar.
[INFO] Including org.codehaus.jackson:jackson-mapper-asl:jar:1.9.13 in the shaded jar.
```

File: flink-connectors/flink-sql-connector-hive-3.1.2/src/main/resources/META-INF/NOTICE (hunk 2 above)
Comment (JingsongLi) [reply flink-connectors/flink-sql-connector-hive-3.1.2/src/main/resources/META-INF/NOTICE:55]: I found it by `unzip -l flink-sql-connector-hive-3.1.2_2.11-1.12-SNAPSHOT.jar | grep pom`...

File: flink-connectors/flink-sql-connector-hive-3.1.2/src/main/resources/META-INF/NOTICE (hunk 4 above)
Comment (JingsongLi) [reply flink-connectors/flink-sql-connector-hive-3.1.2/src/main/resources/META-INF/NOTICE:39]: You are right, I made mistake.

File: flink-connectors/flink-sql-connector-hive-3.1.2/src/main/resources/META-INF/NOTICE (hunk 3 above)
Comment (JingsongLi) [reply flink-connectors/flink-sql-connector-hive-3.1.2/src/main/resources/META-INF/NOTICE:26]: I think you are right, I am confused by the version passed from `flink-parquet`.(Which is from flink pom avro version)

Comment (JingsongLi) [reply flink-connectors/flink-sql-connector-hive-3.1.2/src/main/resources/META-INF/NOTICE:26]: `org.apache.avro:avro-mapred:1.7.7` is enough? Do you we need `jar:hadoop2`?

File: flink-connectors/flink-sql-connector-hive-3.1.2/src/main/resources/META-INF/NOTICE (hunk 5 above)
Comment (JingsongLi) [reply flink-connectors/flink-sql-connector-hive-3.1.2/src/main/resources/META-INF/NOTICE:44]: should come from `parquet-hadoop-bundle`, there are classes in `shaded/parquet/org/codehaus/jackson`.

**Comment by JingsongLi** [comment]
> I've compiled the respective hive versions to compare the output with `hive-exec` shading.
> 
> For Hive 3.1.2 my versions are weirdly different. I guess we used different methods to obtain the dependency list.

For Hive 3.1.2, maybe I was misunderstood by `META-INF/DEPENDENCIES` file.

File: flink-connectors/flink-sql-connector-hive-3.1.2/src/main/resources/META-INF/NOTICE (hunk 3 above)
Comment (rmetzger) [reply flink-connectors/flink-sql-connector-hive-3.1.2/src/main/resources/META-INF/NOTICE]: I was wondering the same. I think it is fine to leave the "hadoop2" part out

File: flink-connectors/flink-sql-connector-hive-3.1.2/src/main/resources/META-INF/NOTICE (hunk 5 above)
Comment (rmetzger) [reply flink-connectors/flink-sql-connector-hive-3.1.2/src/main/resources/META-INF/NOTICE:41]: ok

**Pull request merged**